crates/bls-crypto/src/bls/public.rs

```diff
@@ -43,6 +43,23 @@ impl PublicKey {
             .into()
     }
 
+    /// Computes the dot product of a vector of public keys and a vector of (ideally small or sparse) exponents.
+    pub fn batch(r: &[Fr], public_keys: &[&PublicKey]) -> PublicKey {
+        let projective_elements = public_keys
+            .iter()
+            .map(|pk| pk.as_ref().clone())
+            .collect::<Vec<G2Projective>>();
+
+        let bases = G2Projective::batch_normalization_into_affine(&projective_elements);
```

Comment:
Why do we normalize here?

Reply:
https://docs.rs/ark-ec/0.3.0/ark_ec/msm/struct.VariableBaseMSM.html takes affine bases.

Reply:
Oops, yeah, I realized it before finishing the review but forgot to delete this. Thanks